projects/templates/src/lib/components/po-page-customization/po-page-customization.service.ts

```diff
@@ -0,0 +1,92 @@
+import { HttpClient } from '@angular/common/http';
+import { Observable, from } from 'rxjs';
+import { Injectable } from '@angular/core';
+
+import { PoPageOptions, UrlOrPoCustomizationFunction } from './po-page-options.interface';
+import { map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class PoPageCustomizationService {
+
+  constructor(private http: HttpClient) { }
+
+  getCustomOptions(origin: UrlOrPoCustomizationFunction, originalOption: PoPageOptions): Observable<PoPageOptions> {
+    return this.createObservable(origin).pipe(
+      map(newPageOptions => this.mergePageOptions(originalOption, newPageOptions))
+    );
+  }
+
+  changeOriginalOptionsToNewOptions<T, K>(objectToChange: T, newOptions: K) {
+    Object.keys(newOptions).forEach(key => {
+      const value = newOptions[key];
+      if (objectToChange[key]) {
+        if (Array.isArray(value)) {
+          objectToChange[key] = [...value];
+          return;
+        }
+        if ((typeof (value) === 'number' || typeof (value) === 'string')) {
+          objectToChange[key] = value;
+          return;
+        }
+        if (typeof (value) === 'object') {
+          objectToChange[key] = { ...value };
+        }
+      }
+    });
+  }
+
+  private createObservable(origin: UrlOrPoCustomizationFunction): Observable<PoPageOptions> {
+    if (typeof origin === 'string') {
+      return this.http.post<PoPageOptions>(origin, {});
+    }
+    return from(Promise.resolve(origin()));
```

Comment:
Daria para resolver esse trecho retornando somente 'of(origin())'?

Reply:
Não pq o of vc faria de um objeto interável (como um array) nesse caso eu não sei oque vai voltar da função do usuario então eu uso o from.
Como o from precisa de uma função assincrona ou uma promise eu envelopei ela numa promisse